scripts/gha/test_simulator.py

```diff
@@ -387,23 +389,77 @@ def _build_tvos_helper(helper_project, device_name, device_os):
         return os.path.join(file_dir, file_name)
 
 
-def _record_apple_tests(video_name):
-  command = "xcrun simctl io booted recordVideo -f --codec=h264 %s" % video_name
-  logging.info("Recording test: %s", command)
-  return subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, preexec_fn=os.setsid)
-
-
-def _stop_recording(record_process):
-  logging.info("Stop recording test")
-  try:
-    os.killpg(record_process.pid, signal.SIGINT)
-  except:
-    logging.info("Stop recording test failed!!!")
+def _record_apple_tests(video_name, display):
+  args = [
+    "xcrun",
+    "simctl",
+    "io",
+    "booted",
+    "recordVideo",
+    "-f",
+    "--codec=h264",
+  ]
+  if display is not None:
+    args.append("--display=" + display)
+  args.append(video_name)
+  return _start_recording(args)
+
+
+def _start_recording(args):
+  logging.info("Starting screen recording: %s", subprocess.list2cmdline(args))
+
+  output_file = tempfile.TemporaryFile()
+
+  # Specify CREATE_NEW_PROCESS_GROUP on Windows because it is required for
+  # sending CTRL_C_SIGNAL, which is done by _stop_recording().
+  proc = subprocess.Popen(
+    args,
+    stdout=output_file,
+    stderr=subprocess.STDOUT,
+    creationflags=
+        subprocess.CREATE_NEW_PROCESS_GROUP
+        if platform.system() == "Windows"
```

Comment:
In **CI**, this script only works on **mac** runner. 
It's save to delete it, or you could try it locally and make sure it works on local windows machine.

Reply:
I tried this on all 3 OS's (windows, mac, linux) and the logic works correctly; that is, it saves the screen recordings and is able to upload them as artifacts. Check out https://github.com/firebase/firebase-cpp-sdk/actions/runs/2923963604, which didn't suffer from the Android emulator timeout because I hardcoded the first test to crash.